server/chat-plugins/youtube.ts

```diff
@@ -252,6 +255,23 @@ export class YoutubeInterface {
 		const result = JSON.parse(raw);
 		return result?.items.map((item: AnyObject) => item?.snippet?.channelId);
 	}
+	runInterval(time: string) {
+		let interval = Number(time);
+		if (interval < 10) throw new Chat.ErrorMessage(`${interval} is too low - set it above 10 minutes.`);
+		this.intervalTime = interval;
+		this.data.intervalTime = interval;
+		interval = interval * 60 * 1000;
+		if (this.interval) clearInterval(this.interval);
+		this.interval = setInterval(() => {
+			void (async () => {
+				const room = Rooms.get('youtube');
+				if (!room) return; // do nothing if the room doesn't exist anymore
+				const res = await YouTube.randChannel();
+				room.add(`|html|<div class="infobox">${res}</div>`).update();
+			})();
+		}, interval);
+		return this.interval;
+	}
 }
 
 export const YouTube = new YoutubeInterface(channelData);
```

Comment:
```suggestion
export const YouTube = new YoutubeInterface(channelData);

export function destroy() {
	if (YouTube.interval) clearInterval(YouTube.interval);
}
```

I don't see a cleanup for the interval. Do you need something like this?